### PR TITLE
Infomanager cleanup

### PR DESCRIPTION
**Pull request by ronie**
the documentation in GuiInfoManager regarding `Container` infolabels/infobools was inaccurate (or missing) in some places.
this lead to confusion among skinners which of those labels support the specification of a container id.

i've checked each container infolabel against the code in https://github.com/xbmc/xbmc/blob/master/xbmc/guilib/guiinfo/GUIControlsGUIInfo.cpp to verify if specifying an id was supported or not.

### CODE REVIEW

**Review by enen92 — APPROVED** [review]

**Review by fuzzard — APPROVED** [review]

**Comment by scott967** [comment]
Thanks for this.  I know it's mentioned elsewhere, but do you think it would be worth noting for Container.SortMethod(parameter) that the parameter can (should) be localized string?  I don't know how many instances overall of this there are.

**Comment by ronie** [comment]
@scott967 could you please provide an example?

either it's broken, or i just don't understand how to use it, but it does nothing for me...

**Comment by scott967** [comment]
Sure:

String.IsEqual(Container.SortMethod,$LOCALIZE[551])
vice
String.IsEqual(Container.SortMethod,name)
which only works in English

**Comment by ronie** [comment]
ah ok, please note that's the **infolabel** `Container.SortMethod`
and not the **infobool** `Container.SortMethod(parameter)` you mentioned in your previous post.

i'll update the docs for the infolabel.
i'm still interested to find out how to use the infobool.

**Comment by scott967** [comment]
You.re right.  But it doesn't seem like the boolean is working?  I need it for example in music views with artist sort when system setting is useartistsortname.  For me,  
String.IsEqual(Container.SortMethod,$LOCALIZE[557])  
works but
Container.SortMethod(artist)
doesn't seem to.

**Comment by ronie** [comment]
it was broken indeed. i've added a fix to this PR:
https://github.com/xbmc/xbmc/pull/17667/commits/22c6553f78581fedf129ef163a4e9070e8274397

after staring at the code for hours, i've also figured out how to use it.
you need to provide the [sort id](https://kodi.wiki/view/List_of_built-in_functions#List_of_sort_methods), so for `artist` it would be:
`Container.SortMethod(10)`

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -4042,12 +4052,25 @@ const infomap container_bools[] ={{ "onnext",           CONTAINER_MOVE_NEXT },
...
+///   \table_row3{   <b>`Container.SortMethod`</b>,
+///                  \anchor Container_SortMethod
+///                  _string_,
+///     @return The current sort method (returns a localized value).
+///     <p>
+///   }
+///   \table_row3{   <b>`Container.SortMethod(sortid)`</b>,
+///                  \anchor Container_SortMethod_sortid
+///                  _boolean_,
+///     @return **True** if the current sort method matches the specified SortID (see SortUtils.h).
```
Comment (enen92) [inline xbmc/GUIInfoManager.cpp:4064]: (see \ref List_of_sort_methods "SortUtils")

Comment (ronie) [reply xbmc/GUIInfoManager.cpp]: thanx, updated.

**Comment by phunkyfish** [comment]
jenkins build this please

**Comment by phunkyfish** [comment]
I'll never understand our Jenkins. I created a PR after one and it's in the build queue before it. A mystery...

**Pull request merged**